dbus-serialbattery/bms/daly_can.py

```diff
@@ -37,7 +37,7 @@ def __init__(self, port, baud, address):
         self.poll_step = 0
         self.type = self.BATTERYTYPE
         self.can_bus = None
-        self.device_address = int.from_bytes(address, byteorder="big") if address is not None else 0
+        self.device_address = int.from_bytes(address, byteorder="big") if address is not None else 0x01
```

Comment:
From what I know the address for Daly starts with `0x40` and it should never be empty, since it's provided by the code.

https://github.com/mr-manuel/venus-os_dbus-serialbattery/blob/0c636f731f4fad4997a172558e13f551c93b06b9/dbus-serialbattery/dbus-serialbattery.py#L60-L61

Did you test that with a physical device or with your test script?

Reply:
yes, that definitely has to be clearified. I'm still waiting for the new bms, so I have tested only with my single physical device and my simulator. I have to set the bms to another address to have a real test. give me some days...

additional remarks:
* the address definition that you mentioned above is currently used only for serial, not can
* the meaning of adresses 0x40/0x80 is to be clearified, as the can protocol description sounds the opposite round compared to serial.
![image](https://github.com/user-attachments/assets/e3aa6587-7e7b-4b4c-b72b-fe6909af2b1d)
here, the bms address is 1 and computer=40 and app=80

Reply:
I experimented a bit and found the following relations between configured board no, serial and can addresses:

Board No;serial address;CAN address
1;0x40;0x01
2;0x41;0x02
.
.
.
24;0x57;0x18

The corrent code respects that and is working.  I tested that with a single physical BMS with serial and can connecitons. I will retest with real parallel conneted deviced when I get them, but I think it should work.